csrc/multidevice/utils.cpp

```diff
@@ -60,6 +61,9 @@ std::unordered_set<TensorView*> haveDifferentSharding(
   }
 
   for (auto tv : tvs) {
+    // isSharded asserts that there are no split/merge and that only the outmost
```

Comment:
Actually, let's move the assumptions to the beginning of this function as it also affects how we can use `ComputeAtMap`.

Reply:
Also, shouldn't we check `ref` as well?